activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb

```diff
@@ -1510,6 +1530,25 @@ def valid_primary_key_options # :nodoc:
       end
 
       private
+        MAX_GENERATED_INDEX_NAME_BYTES = 62
```

Comment:
Perhaps not for this PR, but is it possible to ask the adapter what the max length is?  Different DBs have different restrictions, so it makes sense that each adapter declare the max length.